recipes/libboxes/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version
+import os
+
+class libboxesRecipe(ConanFile):
+    name = "libboxes"
+    description = "Boxes is a set of frequently used containers built on top of STL"
+
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://boxes-twdev-projects-ba06c75b02a9b24f40d3a6b9b7130a3266c6a3c532.gitlab.io/"
+
+    topics = ("container", "utility")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "12",
+            "clang": "14",
+            "gcc": "11",
+            "msvc": "192",
+            "Visual Studio": "16.2",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("xxhash/0.8.2", transitive_headers=True)
+        self.test_requires("gtest/1.14.0")
```

Comment:
We really avoid running unit test from the recipes in CCI, the idea packaging the project only, not using Conan as a build tool.

Reply:
Hi @uilianries.  Thanks for the review.  This is something unclear to me so, would appreciate some more details.  Should I disable building and packaging all tests/benchmarks in conanfile.py?  I'm not sure if I should have two versions of conanfile.py - the one submitted to conan-center-index - with no test dependencies and a separate one for local use (whatever that might be) - which explicitly defines all dependencies?  Thanks in advance.